Added support for Date/DateTime Axes

christy-thompson opened the pull request:
Support for Date/DateTime Axes for linear, linear3, bar chart, scatter.

mykelk commented:
It looks like there is a merge conflict that needs to be resolved with the ipynb? Are you able to resolve?

mykelk commented on src/plots.jl:
```diff
@@ -19,8 +20,10 @@ end
 
 abstract type Plot end
 
+XPlotValue = Union{Date, DateTime, Real}
```
Why not just call it PlotValue? Is there a name clash?

christy-thompson replied:
I chose XPlotValue instead because Date and DateTime are not supported for the y, z, etc axis -- only x. But if you think PlotValue is a better name, I'm happy to change!

mykelk commented on src/preamble.tex:
```diff
@@ -1,4 +1,5 @@
 \usepackage{pgfplots}
+\usepgfplotslibrary{dateplot}
```
Let's not change this. Let's use `pushPGFPlotsPreamble` like we do for the polar plots, etc.

christy-thompson replied:
Thank you for the comment -- changed and re-pushed!

mykelk commented:
Are you able to remove Revise? Also, can you drop the adding of packages in the notebook? It is breaking the CI.

christy-thompson commented:
> Are you able to remove Revise? Also, can you drop the adding of packages in the notebook? It is breaking the CI.

I think I already removed Revise, and can't find it anywhere in the current code -- would you mind letting me know where you see it? 
I deleted the "Pkg.add" first cell, but this was already a markdown cell, so not sure if this will solve the issue -- let me know if there's anything else I can do!

mossr commented:
@christy-thompson you're almost there, I think all you need to do is add `Dates` to the `[target]` dependencies for `test`. So chance this in the `Project.toml` file:
```toml
[targets]
test = ["NBInclude", "DataFrames", "Random", "Test"]
```
to this:
```
[targets]
test = ["NBInclude", "DataFrames", "Dates", "Random", "Test"]
```
